pkg/logging/logging.go

```diff
@@ -152,8 +171,10 @@ func setLogLevelFromOptions(logOpts LogOptions) {
 
 // SetupLogging sets up each logging service provided in loggers and configures
 // each logger with the provided logOpts.
-func SetupLogging(loggers []string, logOpts map[string]string, tag string, debug bool) error {
-	logOptions = logOpts
+func SetupLogging(loggers []string, logOpts LogOptions, tag string, debug bool) error {
+	if format, configured := logOpts.GetLogFormat(); configured && format != "text" {
+		DefaultLogger.Formatter = GetFormatter(format)
```

Comment:
Do we sanitize the input here somewhere?

I'm concerned that the new `GetFormatter()` implementation may return `nil` and set the Formatter here in such a way that will later cause a nil dereference and crash. It'd be nicer if we validated that the option is in a known good set first and `log.Fatal(...)` if the user configured it wrong.

Reply:
definitely, I ensured that we always return a TextFormatter and trigger a `log.Error` if the specified format is not supported. what do you think ? 🤔